File: 2.2. Stack/1. Valid Parentheses/dory.py
# Runtime 33ms
# Beats 72.25%

# Memory 16.64MB
# Beats 15.95%
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        dic = {'}': '{',']': '[', ')': '('}
        if len(s) % 2 != 0:
            return False
        for value in s:
            if value in dic.keys():
                if len(stack) == 0:
                    return False
                else:
                    if dic[value] != stack.pop():
                        return False
            else:
                stack.append(value)
        return len(stack) == 0

File: 3.3. Linked List/3. Linked List Cycle/dory.cs
// Runtime 113ms
// Beats 11.99%

// Memory 48.81MB
// Beats 20.96%

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
 
public class Solution {
    public bool HasCycle(ListNode head) {
        HashSet<ListNode> seen = new HashSet<ListNode>();
        ListNode curr = head;

        while(curr != null) {
            if(seen.Contains(curr)) {
                return true;
            }
            seen.Add(curr);
            curr = curr.next;
        }
        return false;
    }
}